### PR TITLE
use generic notanorm uri, instead of file path

### PR DESCRIPTION
**Pull request by earonesty**

### CODE REVIEW

File: policy_basics/per_profile_throttle.py [hunk 1]
```diff
@@ -77,21 +79,29 @@ def increment(self):
...
 class ProfileThrottleDb:
-    db: AbstractDb
+    db: Union[MemoryDb, UriDb]
 
     def __init__(self, args):
         if not args.get("persistent", False):
             self.db = MemoryDb()
         else:
-            path = str(args.get("db-file", os.path.expanduser("~/profile-throttle.db")))
+            uri = args.get("db-uri")
```
Comment (higleyc) [inline policy_basics/per_profile_throttle.py:88]: the table name should be an additional overridable argument

File: policy_basics/simple_db.py [hunk 2]
```diff
@@ -31,23 +41,39 @@ def remove(self, key):
...
-            self.db.execute(
-                "create table if not exists %s (key primary key, val)"
-                % FileDb.TABLE_NAME,
+            self.db = notanorm.open_db(self.uri)
+
+            if self.db.uri_name == "sqlite":
+                self.db.execute("PRAGMA journal_mode=WAL;")
+                self.db.execute("PRAGMA synchronous=NORMAL;")
+
+            self.db.execute_ddl(
```
Comment (higleyc) [inline policy_basics/simple_db.py:71]: if exists?

File: policy_basics/per_profile_throttle.py (hunk 1 above)
Comment (earonesty) [reply policy_basics/per_profile_throttle.py:88]: yep thanks

File: policy_basics/simple_db.py (hunk 2 above)
Comment (earonesty) [reply policy_basics/simple_db.py:74]: ignore_existing defaults to true

**Review by higleyc — APPROVED** [review]

**Pull request merged**